pipes/WDL/workflows/demux_metag.wdl

```diff
@@ -5,18 +5,26 @@ import "tasks_metagenomics.wdl" as metagenomics
 import "tasks_taxon_filter.wdl" as taxon_filter
 import "tasks_assembly.wdl" as assembly
 import "tasks_reports.wdl" as reports
+import "tasks_read_utils.wdl" as reads
 
 workflow demux_metag {
   call demux.illumina_demux as illumina_demux
 
   scatter(raw_reads in illumina_demux.raw_reads_unaligned_bams) {
+      call reads.dedup_bam as dedup {
+          input:
+              in_bam = raw_reads        
+      }
+  }
+
+  scatter(reads_bam in dedup.dedup_bam) {
```

Comment:
I wouldn't double-scatter. You can just keep this as a single scatter block on the raw_reads and put all the task calls together in that single scatter. WDL interpreters/compilers are smart enough to figure out the DAG and parallelization opportunities within the scatter based on the dependencies between their inputs and outputs.

Reply:
So specifically:
```
  scatter(raw_reads in illumina_demux.raw_reads_unaligned_bams) {
    call reads.dedup_bam as dedup {
        input:
            in_bam = raw_reads        
    }
    call reports.spikein_report as spikein {
      input:
        reads_bam = dedup.dedup_bam
    }
    call taxon_filter.deplete_taxa as deplete {
      input:
        raw_reads_unmapped_bam = dedup.dedup_bam
    }
    call assembly.assemble as spades {
      input:
        assembler = "spades",
        reads_unmapped_bam = deplete.cleaned_bam
    }
  }
```

Reply:
Two more thoughts on this topic:
1. Importantly, merging them would allow the execution platform to keep going on the linear DAG portions of each sample as they become ready without waiting for all samples to complete dedup before proceeding to the next steps.
2. I wonder if we should consider running the spikein counting step on raw / non-deduplicated reads.... ERCC's are so short that we might quickly hit an artificial upper bound on the counts if we do it on dedup output.